### ISSUE TITLE
RealmList.move behaves incorrectly if RealmList is unmanaged

### ISSUE THREAD

**Issue by Waboodoo** [realm/realm-java#5860]
#### Goal
I have an unmanaged `RealmList` with multiple elements. I want to update it via the `move(int oldPos, int newPos)` method to notify it that an element has changed positions.

Scenario 1: Move an item to a higher position:
```kotlin
class MyObject(var label: String) : RealmObject()

val myList = RealmList<MyObject>()
myList.add(MyObject("A"))
myList.add(MyObject("B"))

myList.move(0, 1)

assert(myList[0]?.label == "B")
assert(myList[1]?.label == "A")
```

Scenario 2: Move an item to a lower position:
```kotlin
class MyObject(var label: String) : RealmObject()

val myList = RealmList<MyObject>()
myList.add(MyObject("A"))
myList.add(MyObject("B"))

myList.move(1, 0)

assert(myList[0]?.label == "B")
assert(myList[1]?.label == "A")
```

#### Expected Results
In both scenarios, I expect that "A" and "B" have swaped places, i.e., my list is now "B", "A".

#### Actual Results
In scenario 1 the list remains unchanged. (Still "A, "B")
In scenario 2 the items have swaped places as expected.

#### Steps & Code to Reproduce
See my code samples above. The problem only occurs with unmanaged `RealmList`. The items reorder correctly when the list is managed.

#### Code Sample
See above.

#### Version of Realm and tooling
Realm version(s): 5.0.0

Realm sync feature enabled: no

Android Studio version: 3.0.1

Which Android version and device: Android 7.0 on an LG G6

**Comment by Zhuinden** [comment]
Unmanaged `RealmList` should work just like an ArrayList.

Are you sure the behavior differs from the behavior of a regular ArrayList?

**Comment by Waboodoo** [comment]
There is no `move(int oldPos, int newPos)` method on a regular `ArrayList`. The implementation comes directly from the `RealmList` class.

**Comment by Zhuinden** [comment]
Oh yeah, you're right. [Hmm](https://github.com/realm/realm-java/blob/3ee7fa2284a57f790b9e9add52c27c56b8ce9b03/realm/realm-library/src/main/java/io/realm/RealmList.java#L262-L274)...

**Comment by cmelchior** [comment]
We have a unit test for this: https://github.com/realm/realm-java/blob/master/realm/realm-library/src/androidTest/java/io/realm/RealmListTests.java#L318 but I'm able to reproduce your problem, so clearly something is wrong in both implementation and existing test. I'm looking into it

**Comment by cmelchior** [comment]
Fixed in #5872
